gandalf.js

```diff
@@ -6,101 +6,104 @@
  *
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.